airbyte-webapp/src/components/connection/CatalogTree/next/DestinationNamespaceModal.tsx

```diff
@@ -0,0 +1,346 @@
+import { Text } from "components/ui/Text";
+
+import { NamespaceDefinitionType } from "../../../../core/request/AirbyteClient";
+import { FormikConnectionFormValues } from "../../../../views/Connection/ConnectionForm/formConfig";
+import LabeledRadioButton from "../../../LabeledRadioButton";
+import styles from "./DestinationNamespaceModal.module.scss";
+
+interface ExampleSettingsTableProps {
+  columns: Array<{ id: string; displayName: string }>;
+  data: Array<Record<string, string>>;
+}
+
+const ExampleSettingsTable: React.FC<ExampleSettingsTableProps> = ({ columns, data }) => {
+  return (
+    <table className={styles.exampleSettingsTable}>
+      <thead>
+        <tr>
+          {columns.map((column, index) => (
+            <th key={`column_${index}`}>
+              <Text className={styles.text} size="xs">
+                {column.displayName}
+              </Text>
+            </th>
+          ))}
+        </tr>
+      </thead>
+      <tbody>
+        {data.map((row, rowIndex) => (
+          <tr key={`row_${rowIndex}`}>
+            {columns.map((column, dataIndex) => (
+              <td key={`dataIndex_${dataIndex}`}>
+                <Text size="xs">{row[column.id]}</Text>
+              </td>
+            ))}
+          </tr>
+        ))}
+      </tbody>
+    </table>
+  );
+};
+
+const destinationNamespaceValidationSchema = yup.object().shape({
+  namespaceDefinition: yup
+    .string()
+    .oneOf([NamespaceDefinitionType.source, NamespaceDefinitionType.destination, NamespaceDefinitionType.customformat])
+    .required("form.empty.error"),
+  namespaceFormat: yup.string().when("namespaceDefinition", {
+    is: NamespaceDefinitionType.customformat,
+    then: yup.string().trim().required("form.empty.error"),
+  }),
+});
+
+export interface DestinationNamespaceFormValueType {
+  namespaceDefinition: NamespaceDefinitionType;
+  namespaceFormat: string;
+}
+
+interface DestinationNamespaceModalProps {
+  initialValues: Pick<FormikConnectionFormValues, "namespaceDefinition" | "namespaceFormat">;
+  onCloseModal: () => void;
+  onChange: (values: DestinationNamespaceFormValueType) => void;
+}
+
+export const DestinationNamespaceModal: React.FC<DestinationNamespaceModalProps> = ({
+  initialValues,
+  onCloseModal,
+  onChange,
+}) => {
+  const { formatMessage } = useIntl();
+
+  const getInfo = (namespaceDefinition: NamespaceDefinitionType): React.ReactNode => {
+    switch (namespaceDefinition) {
+      case NamespaceDefinitionType.source:
+        return <FormattedMessage id="connectionForm.modal.destinationNamespace.option.source.description" />;
+      case NamespaceDefinitionType.destination:
+      case NamespaceDefinitionType.customformat:
+        return <FormattedMessage id="connectionForm.modal.destinationNamespace.option.destination.description" />;
+    }
+  };
+
+  const getExampleTableData = (namespaceDefinition: NamespaceDefinitionType): ExampleSettingsTableProps => {
```

Comment:
This could be extracted out of the component function and into its own, same for `getInfo`

Reply:
But I have a dependency with `formatMessage` for those functions :worried: